#! /usr/bin/env python
# Honeybee dynamic configuration

PACKAGE='arm_honeybee'
import roslib; roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *
from dynamic_reconfigure.msg import SensorLevels

gen = ParameterGenerator()

# Name, Type, Reconfiguration level, Description, Default, Min, Max
gen.add("fps", double_t, SensorLevels.RECONFIGURE_CLOSE,
         "Camera frames per second.", 15.0, 1.875, 15.0)

gen.add("auto_brightness", bool_t, SensorLevels.RECONFIGURE_RUNNING,
        "Camera auto brightness.", True)

gen.add("brightness", double_t, SensorLevels.RECONFIGURE_RUNNING,
        "Camera brightness setting.", 0.000, 0.000, 6.225)

gen.add("auto_exposure", bool_t, SensorLevels.RECONFIGURE_RUNNING,
        "Camera auto exposure.", True)

gen.add("exposure", int_t, SensorLevels.RECONFIGURE_RUNNING,
        "Camera auto exposure setting.", 1, 1, 1023)

gen.add("auto_gain", bool_t, SensorLevels.RECONFIGURE_RUNNING,
        "Camera auto gain.", True)

gen.add("gain", double_t, SensorLevels.RECONFIGURE_RUNNING,
        "Camera gain setting.", 0.000, 0.000, 24.035)

gen.add("auto_shutter", bool_t, SensorLevels.RECONFIGURE_RUNNING,
        "Camera auto shutter.", True)

gen.add("shutter", double_t, SensorLevels.RECONFIGURE_RUNNING,
        "Camera shutter setting.", 0.0001, 0.0001, 0.0666)

gen.add("whitebalance", str_t, SensorLevels.RECONFIGURE_RUNNING,
        "Sets the Blue/U and Red/V components of white balance.", "710 524")

exit(gen.generate(PACKAGE, "arm_honeybee", "honeybee"))
